src/app/header/header.component.spec.ts

```diff
@@ -1,25 +1,57 @@
+import { Menu } from './../shared/menu/menu.interface';
+import { registeredMenu, unRegisteredMenu } from './../shared/menu/menu.lists';
+import { MenuService } from './../shared/menu/menu.services';
+import { User } from './../shared/users/user.interface';
+import { userMock } from './../shared/users/user.mocked';
+import { UserService } from './../shared/users/user.services';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { HeaderComponent } from './header.component';
 
 describe('HeaderComponent', () => {
   let component: HeaderComponent;
   let fixture: ComponentFixture<HeaderComponent>;
+  let userServiceStub: Partial<UserService>;
+  let menuServiceStub: Partial<MenuService>;
+  let userService: UserService;
+  let menuService: MenuService;
 
   beforeEach(async () => {
+    userServiceStub = {
+      user: JSON.parse(JSON.stringify(userMock))
+    };
+    menuServiceStub = {
+      registeredMenu,
+      unRegisteredMenu,
+      getMenu(isUserLogin: boolean): Menu[] {
+        return isUserLogin ? this.registeredMenu : this.unRegisteredMenu;
+      }
+    };
```

Comment:
The more convenient way is to create `menu|user.service.mock.ts` and define services' mocks and providers. this will allow you to re-use them in other tests that depend on this mocks

Reply:
done